sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OffHeapColumnVector.java

```diff
@@ -64,13 +64,23 @@
   private long lengthData;
   private long offsetData;
 
+  // Only set if type is Struct.
+  //
+  // A slot 'structOffsetData[i]' is only defined iff 'nulls[i]' is NOT set. If defined,
```

Comment:
This comment is a bit confusing. `structOffsetData` is a long and what does `structOffsetData[i]` mean?

Reply:
oh, it's an off-heap int array.

Reply:
actually "defined" is a bit confusing when I first read this, because it doesn't look like we set it in "undefined" value. I guess its meaning is, if the index i is null at this column vector, then the index of `structOffsetData` is meaningful and will/can be used.

Reply:
Hmm when `nulls[I]` is set, `structOffsetData[i]` is literally undefined since it can be any value right? the memory region is uninitialized.

Reply:
yea, I will think "defined" in the context of `Option`-like class that you can definitely know the value is defined or not. It doesn't apply for here as it is just uninitialized so we cannot tell if a value `structOffsetData[i]` is "defined" or not by just looking at its value.

Reply:
I can try to revise it if it sounds confusing. How about: 

> A slot 'structOffsetData[i]' will only be set iff 'nulls[i]' is NOT set. If set, 'structOffsetData[i] = j' indicates that for struct at slot 'i', its offset in the child vectors is 'j'. For those slots that are not set, their values are undefined and should never be used.

Reply:
looks better.